src/Sidebar/ui/SidebarRoot.tsx

```diff
@@ -59,6 +55,39 @@ import { InvitationsSeen } from "../../Faction/ui/FactionsRoot";
 import { hash } from "../../hash/hash";
 import { Locations } from "../../Locations/Locations";
 
+// All icon instances need to be constant, so they have stable object identity.
+// Otherwise, the memoization of all the higher-level components doesn't work.
+const computerIcon = <ComputerIcon />;
+const lastPageIcon = <LastPageIcon />;
+const createIcon = <CreateIcon />;
+const storageIcon = <StorageIcon />;
+const bugReportIcon = <BugReportIcon />;
+const equalizerIcon = <EqualizerIcon />;
+const contactsIcon = <ContactsIcon />;
+const doubleArrowIcon = <DoubleArrowIcon style={{ transform: "rotate(-90deg)" }} />;
+const accountTreeIcon = <AccountTreeIcon />;
+const peopleAltIcon = <PeopleAltIcon />;
+const locationCityIcon = <LocationCityIcon />;
+const airplanemodeActiveIcon = <AirplanemodeActiveIcon />;
+const workIcon = <WorkIcon />;
+const trendingUpIcon = <TrendingUpIcon />;
+const formatBoldIcon = <FormatBoldIcon />;
+const businessIcon = <BusinessIcon />;
+const sportsMmaIcon = <SportsMmaIcon />;
+const checkIcon = <CheckIcon />;
+const helpIcon = <HelpIcon />;
+const settingsIcon = <SettingsIcon />;
+const developerBoardIcon = <DeveloperBoardIcon />;
+const emojiEventsIcon = <EmojiEventsIcon />;
+const accountBoxIcon = <AccountBoxIcon />;
+const publicIcon = <PublicIcon />;
+const liveHelpIcon = <LiveHelpIcon />;
+const chevronLeftIcon = <ChevronLeftIcon color="primary" />;
+const chevronRightIcon = <ChevronRightIcon color="primary" />;
```

Comment:
Why should the root know about the icons of its children?
Wouldn't it be much better, if it instead gave something the children could use to grab it themselves?
It would stop any rerenders to re-define all of these values, but instead each child that needs to be rerendered grabs it for themselves.
One other way of doing it, would be to put each accordion in an useMemo hook and put the icon in there.
Yes, it would be a bit harder to read and changing the icons would be not as straight forward, but it would stop any unwanted rerenders, because the reference to the icon element changed.

Reply:
Well, the root has to define the data in *some* format - it is the only component that is not "generic," i.e. that knows the concrete data to assign to each place. Using actual ReactElements is very convenient, because it allows for the root to customize the icons in arbitrary ways. This is needed for the Augmentations icon, which has a transform on it.

No changes are needed here to avoid rerenders: There basically aren't any that aren't completely necessary.
![image](https://user-images.githubusercontent.com/459704/205485903-0747dd66-bd86-408a-b712-f4cb6785aa0a.png)

It took me a *long* time to fully wrap my head around how React works with children elements, in order to get this straight. Because the icons are always the same objects when being passed *in* to SidebarItem, `memo` will see this and allow them to be memoized. This means that all the rendering from SidebarItem down gets eliminated, *even though* it creates new versions of the icon elements internally.

The stuff *above* SidebarItem can't really be memoized. (It could, potentially, but only with a ton of effort and it's not really worth it.) In particular, SidebarRoot has to do various calculations and create a fresh array of items to pass to each SidebarAccordian; trying to express that dependency structure would be a really big pain either for `memo` or `useMemo`. And as you can see, the total time taken up by each `SidebarAccordian` render is now very small (although collectively, they are a big chunk of the remaining time).